From c29ac9ed6e87a47397222fd4f9f36d9f5dbdead5 Mon Sep 17 00:00:00 2001
From: "Furtak, Pawel" <pawel.furtak@intel.com>
Date: Thu, 21 Mar 2019 14:13:30 -0800
Subject: [PATCH 1364/1443] ASoC: Intel: Skl: Virt: FE: snd card registration
 notification

Using FE funcion on machine driver registration will be more
generic and will allow to be easily extendable for future
features.

Change-Id: Ic0c438fb28fbb13934cab89a091705fe201848a6
Signed-off-by: Pawel Furtak <pawel.furtak@intel.com>
---
 sound/soc/intel/skylake/virtio/skl-virtio-card.c | 2 +-
 sound/soc/intel/skylake/virtio/skl-virtio-fe.c   | 6 ++++++
 sound/soc/intel/skylake/virtio/skl-virtio.h      | 1 +
 3 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/sound/soc/intel/skylake/virtio/skl-virtio-card.c b/sound/soc/intel/skylake/virtio/skl-virtio-card.c
index fa17f67..baf37a4 100644
--- a/sound/soc/intel/skylake/virtio/skl-virtio-card.c
+++ b/sound/soc/intel/skylake/virtio/skl-virtio-card.c
@@ -341,7 +341,7 @@ static int skl_virtio_card_probe(struct platform_device *pdev)
 	if (ret < 0)
 		return ret;
 
-	kctl_notify_machine_ready(card);
+	vfe_notify_machine_ready(card);
 	return ret;
 }
 
diff --git a/sound/soc/intel/skylake/virtio/skl-virtio-fe.c b/sound/soc/intel/skylake/virtio/skl-virtio-fe.c
index 20a37d4..ae16126 100644
--- a/sound/soc/intel/skylake/virtio/skl-virtio-fe.c
+++ b/sound/soc/intel/skylake/virtio/skl-virtio-fe.c
@@ -806,6 +806,12 @@ static int vfe_platform_register(struct snd_skl_vfe *vfe, struct device *dev)
 	return result;
 }
 
+void vfe_notify_machine_ready(struct snd_soc_card *card)
+{
+	kctl_notify_machine_ready(card);
+}
+EXPORT_SYMBOL(vfe_notify_machine_ready);
+
 static int vfe_machine_device_register(struct snd_skl_vfe *vfe, struct skl *skl)
 {
 	struct snd_soc_acpi_mach *mach = skl->mach;
diff --git a/sound/soc/intel/skylake/virtio/skl-virtio.h b/sound/soc/intel/skylake/virtio/skl-virtio.h
index d2a0311..00b0cee 100644
--- a/sound/soc/intel/skylake/virtio/skl-virtio.h
+++ b/sound/soc/intel/skylake/virtio/skl-virtio.h
@@ -22,6 +22,7 @@ extern void vfe_sst_dsp_cleanup(struct device *dev, struct skl_sst *ctx);
 
 #endif
 
+void vfe_notify_machine_ready(struct snd_soc_card *card);
 void kctl_notify_machine_ready(struct snd_soc_card *card);
 
 #endif //__SOUND_SOC_SKL_VIRTIO_H
-- 
2.7.4

